src/main/scala/io/citrine/lolo/trees/splits/MultiTaskSplitter.scala

```diff
@@ -0,0 +1,166 @@
+        val pivot = (features(j) + features(j - 1)) / 2.0
+        Some((pivot, totalImpurity))
+      } else {
+        None
+      }
+    }
+    if (pivots.isEmpty) {
+      (new NoSplit, Double.MaxValue)
+    } else {
+      val best = pivots.minBy(_._2)
+      (new RealSplit(index, best._1), best._2)
+    }
+  }
+
+  /**
+    * Get find the best categorical splitter.
+    *
+    * @param data        to split
+    * @param totalWeight Pre-computed data.map(d => d._3).sum
+    * @param index       of the feature to split on
+    * @return the best split of this feature
+    */
+  def getBestCategoricalSplit(
+                               data: Seq[(Vector[AnyVal], Array[AnyVal], Double)],
+                               totalWeight: Double,
+                               index: Int,
+                               minCount: Int
+                             ): (Split, Double) = {
+    /* Extract the features at the index */
+    val thinData = data.map(dat => (dat._1(index).asInstanceOf[Char], dat._2, dat._3))
+
+    /* Group the data by categorical feature and compute the weighted sum and sum of the weights for each */
+    val groupedData: Map[Char, (Double, Double, Double)] = thinData.groupBy(_._1).mapValues(g =>
+      (computeImpurity(g.map(x => (x._2, x._3))), g.map(_._3).sum, g.size)
+    )
+
+    /* Make sure there is more than one member for most of the classes */
+    val nonTrivial: Double = groupedData.filter(_._2._3 > 1).map(_._2._2).sum
+    if (nonTrivial / totalWeight < 0.25) {
+      return (new NoSplit, Double.MaxValue)
+    }
+
+    /* Create an orderd list of the categories by average label */
+    val orderedNames: Seq[Char] = groupedData.toSeq.sortBy(_._2._1).map(_._1)
+
+    /* Add the categories one at a time in order of their average label */
+    val pivots = (1 until orderedNames.size).flatMap { j =>
+      val (left, right) = thinData.partition(x => orderedNames.take(j).contains(x._1))
+
+      if (left.size >= minCount && right.size >= minCount) {
+        val totalImpurity = Seq(left, right).map(part => computeImpurity(part.map(x => (x._2, x._3)))).sum
+        val set = orderedNames.take(j).toSet
+        Some(set, totalImpurity)
+      } else {
+        None
+      }
+    }
+    if (pivots.isEmpty) {
+      (new NoSplit, Double.MaxValue)
+    } else {
+      val best = pivots.minBy(_._2)
+      (new CategoricalSplit(index, new scala.collection.mutable.BitSet() ++ best._1.map(_.toInt)), best._2)
+    }
+  }
+
+  /**
+    * Compute the impurity of a set of weighted labels
+    * @param labels is a seq of (Array of multiple labels, single weight)
+    * @return the impurity, which is in [0, number of labels * sum of weights]
+    */
+  def computeImpurity(labels: Seq[(Array[AnyVal], Double)]): Double = {
+    // Return early if there is no impurity
+    if (labels.size == 1 || labels.head._1.isEmpty || labels.map(_._2).sum == 0.0) return 0.0
+
+    // Sum the impurity of each individual label
```

Comment:
Is some normalization required if the labels vary over different ranges?

Reply:
Yes, but I'd like to leave that up to the user.  They can used the standardizer for the labels if they want.  I'll add a multitask standardizer